influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,283 @@
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query, @Nonnull String database,
+                                               @Nonnull SupplierEx<InfluxDB> connectionSupplier,
+                                               @Nonnull Class<T> clazz, int chunkSize) {
+        checkTrue(query != null, "query cannot be null");
+        checkTrue(database != null, "database cannot be null");
+        checkTrue(connectionSupplier != null, "username cannot be null");
+        checkTrue(clazz != null, "clazz cannot be null");
+
+        return SourceBuilder.batch("influxdb-" + database,
+                ignored -> new InfluxDbSource<>(query, database, chunkSize, connectionSupplier, clazz, null))
+                .<T>fillBufferFn(InfluxDbSource::addToBufferWithPOJOMapping)
+                .destroyFn(InfluxDbSource::close)
+                .build();
+    }
+
+    /**
+     * A streaming source which executes a query on InfluxDb and emits
+     * results as they arrive.
+     *
+     * @param <T> emitted item type
+     */
+    private static class InfluxDbSource<T> {
+
+        private final Class<T> clazz;
+        private final BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>(10000);
+        private final ArrayList<QueryResult> buffer = new ArrayList<>();
+        private final InfluxDBResultMapper resultMapper;
+        private final QuadFunction<String, Map<String, String>, List<String>, List<Object>, T> measurementMapper;
+        private InfluxDB db;
+        private volatile boolean finished;
+
+        InfluxDbSource(@Nonnull String query, @Nonnull String database, int chunkSize,
+                       @Nonnull SupplierEx<InfluxDB> connectionSupplier, @Nullable Class<T> clazz,
+                       @Nullable QuadFunction<String, Map<String, String>, List<String>, List<Object>, T>
+                                        measurementMapper) {
+            this.clazz = clazz;
+            this.resultMapper = clazz != null ? new InfluxDBResultMapper() : null;
+            this.measurementMapper = measurementMapper;
+            db = connectionSupplier.get();
+            db.query(new Query(query, database),
+                    chunkSize,
+                    queue::add,
+                    () -> finished = true
+            );
+        }
+
+        void addToBufferWithPOJOMapping(SourceBuffer<T> sourceBuffer) {
+            transferTo(result -> {
+                throwExceptionIfResultWithErrorOrNull(result);
+                if (!result.hasError()) {
+                    resultMapper.toPOJO(result, clazz)
+                                .forEach(sourceBuffer::add);
+                }
+            }, sourceBuffer);
+        }
+
+        void addToBufferWithMeasurementMapping(SourceBuffer<T> sourceBuffer) {
+            transferTo(result -> {
+                throwExceptionIfResultWithErrorOrNull(result);
+                if (!result.hasError()) {
+                    result.getResults()
```

Comment:
the indentation here should be fixed and code refactored to make it easier to read